src/CarbonAware.DataSources/CarbonAware.DataSources.Json/src/JsonDataSource.cs

```diff
@@ -88,7 +88,8 @@ private IEnumerable<EmissionsData> FilterByDateRange(IEnumerable<EmissionsData>
     {
         var (newStartTime, newEndTime) = IntervalHelper.ExtendTimeByWindow(startTime, endTime, MinSamplingWindow);
         var windowData = data.Where(ed => ed.TimeBetween(newStartTime, newEndTime));
-        var filteredData = IntervalHelper.FilterByDuration(windowData, startTime, endTime, TimeSpan.FromHours(DURATION));
+        var duration = GetDurationFromEmissionDataPointsOrDefault(windowData, TimeSpan.FromHours(DURATION));
+        var filteredData = IntervalHelper.FilterByDuration(windowData, startTime, endTime, duration);
```

Comment:
Any reason to not resolve this by removing `duration` as an argument entirely and letting `FilterByDuration` just use the duration of the emissions data points?

These additional helper methods used by the `WattTimeDataSource` get used in a context where the operator is not in control of the underlying WattTime data, which is not true for the JSON files.  I would prefer that we be more opinionated about the structure of the json data file and simply expect a duration value for all data points.